### PR TITLE
Weblogic

### PR DESCRIPTION
**Pull request by huangjien**
add weblogic role

### CODE REVIEW

File: roles/weblogic/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+domain_username: 'weblogic'
+domain_password: 'Password1'
+encrypted_password: 'dummypassword'
+server_name: 'CuramServer'
+server_port: 7001
+# Artifactory info
+artifactory_token: "{{ lookup('env', 'ARTIFACTORY_TOKEN') }}"
+download_url: "{{ lookup('env','ARTIFACTORY_URL') }}/{{ lookup('env','ARTIFACTORY_REPO') }}/SoftwareInstallers"
+# artifactory_url: https://eu.artifactory.swg-devops.com/artifactory
+artifactory_repo: wh-spmdevops-generic-local
```
Comment (salob) [inline roles/weblogic/defaults/main.yml:24]: please remove artifactory information

File: roles/weblogic/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+artifactory_patch_folder: 'Patches'
```
Comment (salob) [inline roles/weblogic/defaults/main.yml:25]: remove

File: roles/weblogic/meta/main.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+---
+galaxy_info:
+  author: SPM Development
```
Comment (salob) [inline roles/weblogic/meta/main.yml:3]: I think we use SPM DevOps Team or something, check one of the other migrated roles

File: roles/weblogic/tasks/install.yml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    owner: "{{ weblogic_user }}"
+    group: "{{ weblogic_group }}"
+
+# - name: Download installer
+#   get_url:
+#     url: "{{ artifactory_url }}/{{ artifactory_repo }}/SoftwareInstallers/WLS/{{ base_install_file }}"
+#     dest: "{{ mw_installer_folder }}/"
+#     headers:
+#       X-JFrog-Art-Api: "{{ artifactory_token }}"
+#   register: ora_installer
```
Comment (salob) [inline roles/weblogic/tasks/install.yml:16]: remove comments please

File: roles/weblogic/tasks/install.yml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  copy:
+    src: "{{ weblogic_installer_path }}"
+    dest: "{{ mw_installer_folder }}/"
+  register: ora_installer
+  when: download_url is not defined
+
+- name: Download weblogic base installer
+  get_url:
+    dest: "{{ mw_installer_folder }}/"
+    url: "{{ download_url }}/WLS/{{ base_install_file }}"
```
Comment (salob) [inline roles/weblogic/tasks/install.yml:28]: I want `weblogic_installer_path` to be used for both get_url and copy, can you please do this way like others:

`weblogic_installer_path: WLS`

then

` url: "{{ download_url }}/{{ weblogic_installer_path }}/{{ base_install_file }}"`

File: roles/weblogic/tasks/patch.yml [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    - name: Create opatch folder
+      file: state=directory path="{{ mw_installer_folder }}/Patches" owner="{{ weblogic_user }}" group="{{ weblogic_group }}"
+
+    # - name: download patch package
+    #   get_url:
+    #     url: "{{ artifactory_url }}/{{ artifactory_repo }}/SoftwareInstallers/WLS/{{ artifactory_patch_folder }}/{{ opatch }}"
+    #     dest: "{{ mw_installer_folder }}/"
+    #     headers:
+    #       X-JFrog-Art-Api: "{{ artifactory_token }}"
+
```
Comment (salob) [inline roles/weblogic/tasks/patch.yml:16]: remove comments please

File: roles/weblogic/tasks/patch.yml [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    - name: Copy installer
+      copy:
+        src: "{{ weblogic_installer_path }}"
+        dest: "{{ mw_installer_folder }}/"
+      when: download_url is not defined
+
+    - name: Download opatch installer
+      get_url:
+        dest: "{{ mw_installer_folder }}/"
+        url: "{{ download_url }}/WLS/{{ artifactory_patch_folder }}/{{ opatch }}"
```
Comment (salob) [inline roles/weblogic/tasks/patch.yml:26]: same as before use `weblogic_installer_path` same as copy

File: roles/weblogic/tasks/patch.yml [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+    #     url: "{{ artifactory_url }}/{{ artifactory_repo }}/SoftwareInstallers/WLS/{{ artifactory_patch_folder }}/{{ opatch }}"
+    #     dest: "{{ mw_installer_folder }}/"
+    #     headers:
+    #       X-JFrog-Art-Api: "{{ artifactory_token }}"
+
+    - name: Copy installer
+      copy:
+        src: "{{ weblogic_installer_path }}"
+        dest: "{{ mw_installer_folder }}/"
+      when: download_url is not defined
```
Comment (salob) [inline roles/weblogic/tasks/patch.yml:21]: should you not also add mode like the get_url ?

File: roles/weblogic/tasks/patch.yml [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+
+    - name: Create patch folder
+      file: state=directory path="{{ mw_installer_folder }}/patch" owner="{{ weblogic_user }}" group="{{ weblogic_group }}"
+
+    # - name: Download patch package
+    #   get_url:
+    #     url: "{{ artifactory_url }}/{{ artifactory_repo }}/SoftwareInstallers/WLS/{{ artifactory_patch_folder }}/{{ patch }}"
+    #     dest: "{{ mw_installer_folder }}/patch"
+    #     headers:
+    #       X-JFrog-Art-Api: "{{ artifactory_token }}"
```
Comment (salob) [inline roles/weblogic/tasks/patch.yml:72]: remove comments

File: roles/weblogic/tasks/patch.yml [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+    - name: Copy installer
+      copy:
+        src: "{{ weblogic_installer_path }}"
+        dest: "{{ mw_installer_folder }}/patch"
+      when: download_url is not defined
+
+    - name: Download patch installer
+      get_url:
+        dest: "{{ mw_installer_folder }}/patch"
+        url: "{{ download_url }}/WLS/{{ artifactory_patch_folder }}/{{ patch }}"
```
Comment (salob) [inline roles/weblogic/tasks/patch.yml:83]: same as before url should use `weblogic_installer_path` if possible

File: roles/weblogic/defaults/main.yml [hunk 11]
```diff
@@ -0,0 +1,26 @@
+---
+weblogic_home: '/home/oracle/Oracle/Middleware/Oracle_Home'
+weblogic_user: 'oracle'
+weblogic_group: 'oinstall'
+weblogic_os_user_pass: 'Password1'
```
Comment (salob) [inline roles/weblogic/defaults/main.yml:5]: perhaps just oracle here

File: roles/weblogic/defaults/main.yml [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+# If not passed in, use this default value
+weblogic_version: '14.1.1.0.210716'
+inv_loc: '/opt/Oracle/oraInventory'
+jdk_folder: '/usr/java'
+# Common paths and names
+domains_home: '{{ weblogic_base }}/config/Domains'
+applications_home: '{{ weblogic_base }}/config/applications'
+mw_installer_folder: '{{ weblogic_base }}/installer'
+domain_username: 'weblogic'
+domain_password: 'Password1'
```
Comment (salob) [inline roles/weblogic/defaults/main.yml:16]: perhaps just weblogic here

**Comment by romakarol** [comment]
There's a good few comments left in, I'd suggest CTRL-F '#' in your changeset tab on this PR and address them

File: roles/weblogic/tasks/patch.yml (hunk 6 above)
Comment (huangjien) [reply roles/weblogic/tasks/patch.yml]: removed

File: roles/weblogic/tasks/patch.yml [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+
+- name: patch weblogic
+  block:
+
+    - name: Create patch folder
+      file: state=directory path="{{ mw_installer_folder }}/patch" owner="{{ weblogic_user }}" group="{{ weblogic_group }}"
+
+    - name: Copy installer
+      copy:
+        src: "{{ weblogic_local_installer_path }}/{{ weblogic_installer_path }}/{{ weblogic_patch_path }}/{{ patch }}"
```
Comment (salob) [inline roles/weblogic/tasks/patch.yml:63]: why the extra var? was this added since our review ?

**Pull request merged**